functions/transformations/sample_test.go

```diff
@@ -30,46 +30,6 @@ func TestSample_Process(t *testing.T) {
 		want   [][]int
 		fromor *transformations.SampleSelector
 	}{
-		{
```

Comment:
These tests seem important. I had begun this work but hadn't finished it and I think we may need to make it easy to test things like a column reader with multiple calls since it will happen.

This is asking a lot and I know, but it's also work that we'll have to do at some point. Can you create a testing utility method to break up.

I think this can be done by figuring out how slicing works. The arrow library supports it, but I'm not 100% on when `Retain()` or `Release()` calls are required. I think this is a rough way of doing it.

```
func BoolSlice(vs *array.Boolean, i, j int) *array.Boolean {
    data := array.NewSliceData(vs.Data(), i, j)
    return array.NewBooleanData(data)
}
```

I think if you add that to our `arrow` utility package, that will work for data slices. It needs to be released after use,  but I'm not sure any other calls are needed.